scripts/skills/perks/perk_rf_bolster.nut

```diff
@@ -15,38 +16,33 @@ this.perk_rf_bolster <- ::inherit("scripts/skills/skill", {
 		this.m.IsHidden = false;
 	}
 
-	function isEnabled()
+	function onAnySkillExecuted( _skill, _targetTile, _targetEntity, _forFree )
 	{
-		if (this.m.IsForceEnabled) return true;
-
-		if (this.getContainer().getActor().isDisarmed()) return false;
+		local actor = this.getContainer().getActor();
+		if (!actor.isPlacedOnMap() || actor.isEngagedInMelee() || !this.isSkillValid(_skill))
+			return;
 
-		local weapon = this.getContainer().getActor().getMainhandItem();
-		if (weapon == null || weapon.getReach() < 6)
+		foreach (ally in ::Tactical.Entities.getFactionActors(actor.getFaction(), actor.getTile(), 1, true))
 		{
-			return false;
+			local moraleState = ally.getMoraleState();
+			if (moraleState < ::Const.MoraleState.Confident)
+			{
+				ally.checkMorale(1, ::Const.Morale.RallyBaseDifficulty);
+				if (ally.getMoraleState() > moraleState)
+					this.spawnIcon("perk_rf_bolster", ally.getTile());
+			}
 		}
-
-		return true;
 	}
 
-	function onAnySkillExecuted( _skill, _targetTile, _targetEntity, _forFree )
+	function isSkillValid( _skill )
 	{
-		if (_targetEntity != null && this.getContainer().getActor().isPlacedOnMap() && !this.getContainer().getActor().isEngagedInMelee() && _skill.isAttack() && !_skill.isRanged() && (this.m.IsForceEnabled || _skill.m.IsWeaponSkill) && this.isEnabled())
-		{
-			local allies = ::Tactical.Entities.getFactionActors(this.getContainer().getActor().getFaction(), this.getContainer().getActor().getTile(), 1, true);
-			foreach (ally in allies)
-			{
-				local moraleState = ally.getMoraleState();
-				if (moraleState < ::Const.MoraleState.Confident)
-				{
-					ally.checkMorale(1, ::Const.Morale.RallyBaseDifficulty);
-					if (ally.getMoraleState() > moraleState)
-					{
-						this.spawnIcon("perk_rf_bolster", ally.getTile());
-					}
-				}
-			}
-		}
+		if (_skill.isRanged() || !_skill.isAttack())
+			return false;
+
+		if (this.m.RequiredWeaponType == null)
```

Comment:
If this is the only time you use `RequiredWeaponType` you could also change it into a bool and call it `RequiresWeapon`.

Otherwise you probably want to include an additional check for this variable in the big return condition below